### ISSUE TITLE
Request for making several elements valid in mei-neume dev-5.0.0 schema

### ISSUE THREAD

**Issue by yinanazhou** [music-encoding/music-encoding#968]
As shown in the figure below, the underlined elements are not valid in the mei-neume dev-5.0.0 schema. The details and examples are as follows. All the examples are from Einsiedeln (CH-E_611).

![mei-Neumes   dev copy](https://user-images.githubusercontent.com/74484576/169932914-70b01afa-afa1-47a3-b293-ed637cea5ed8.png)

1. `<accid>` is not valid both inside and outside a syllable, but it happens both inside and outside.

inside:
<img width="970" alt="018 flat inside 1 - folio" src="https://user-images.githubusercontent.com/74484576/169933659-512e74d9-fb95-432d-8236-2bd1036f163e.png">
<img width="856" alt="018r flat inside 1 - mei" src="https://user-images.githubusercontent.com/74484576/169934080-349709e2-39d9-417b-a2ed-a651a4df5d85.png">

<img width="960" alt="018r flat inside 2 - folio" src="https://user-images.githubusercontent.com/74484576/169933974-26541b54-9168-4d1d-978d-d40c1180975a.png">
<img width="883" alt="018r flat inside 2 - mei" src="https://user-images.githubusercontent.com/74484576/169934107-cc07609c-8cf0-41c2-8f37-8ab017374e46.png">

outside:
<img width="1010" alt="260r flat outside - folio" src="https://user-images.githubusercontent.com/74484576/169933998-7ca3c1b2-6bbd-4c73-ab48-8c9a08c0d0e6.png">
<img width="865" alt="260r flat outside - mei" src="https://user-images.githubusercontent.com/74484576/169934137-dec55c06-0d07-4059-be67-0875e1284cca.png">

<img width="850" alt="261v flat outside - folio" src="https://user-images.githubusercontent.com/74484576/169934029-7ef5bc32-faa6-4893-be64-4ecf1f9f0742.png">
<img width="833" alt="261v flat outside - mei" src="https://user-images.githubusercontent.com/74484576/169934157-2c75eaa8-30d0-43f1-9d02-fbaa55046e41.png">

2. `<divLine>` is only valid inside a syllable, but it is mostly outside.

<img width="960" alt="044r divisio outside - folio" src="https://user-images.githubusercontent.com/74484576/169934987-0c12a4a7-d0fa-474a-b312-eeac6d420388.png">
<img width="847" alt="044r divisio outside - mei" src="https://user-images.githubusercontent.com/74484576/169935494-1ff12114-3c9c-4e2c-9857-6936fefba6da.png">

<img width="977" alt="267v divisios outside - folio" src="https://user-images.githubusercontent.com/74484576/169935010-9d957466-c308-4dde-9869-11d90af47dee.png">
<img width="880" alt="267v divisios outside - mei" src="https://user-images.githubusercontent.com/74484576/169935508-311d0d1c-9cfe-4556-952e-c03e6d9ccd50.png">

3. `<clef>`is only valid outside a syllable, but sometimes it is inside.

<img width="896" alt="270v clef inside 1 and 2 - folio" src="https://user-images.githubusercontent.com/74484576/169935458-f6043225-7083-4bc5-ae2c-c874e130e160.png">
<img width="883" alt="270v clef inside 1 and 2 - mei" src="https://user-images.githubusercontent.com/74484576/169935462-60c6c2ba-cc41-4609-acf2-a3da14083071.png">

<img width="948" alt="061v clef inside - folio" src="https://user-images.githubusercontent.com/74484576/169935346-47c64157-b4f6-4b96-b6fb-69f51ad21c2c.png">
<img width="843" alt="061v clef inside - mei" src="https://user-images.githubusercontent.com/74484576/169935376-2e9c6d6e-86b5-4b1b-835f-a7e768c72818.png">

Compared to mei-all dev-5.0.0 schema, all the elements mentioned above but `<accid>` (also needs to be fixed) are valid both inside and outside a syllable. 

![mei-all   dev copy](https://user-images.githubusercontent.com/74484576/169939749-28e53786-319e-42df-bc48-e758197220f9.png)

**Comment by yinanazhou** [comment]
Thanks to @martha-thomae for testing the schema, and thanks to @JoyfulGen for providing the examples.

**Comment by bwbohl** [comment]
Many thanks for bringing this up. As I'm no expert in encoding neumes I would like to bring this issue to the attention of the @music-encoding/ig-neumes with a kind request of reviewing and discussing the issues mentioned.

**Comment by musicEnfanthen** [comment]
Just for the record: The `clef` and `custos` elements have been disabled within syllable in this commit: 78784fa92208ad7683d0c456840ca45ce326efd8 (maybe @kepper remembers about the intention behind that change?)

`accid` seems never to have been allowed, not even in v4?

Just out of curiosity: Is there a semantic difference between a `clef` or `accid` outside a syllable vs. inside a syllable? If they appear at the beginning of a syllable (that is "outside", right?), would they not be taken to belong to that syllable?

**Comment by fujinaga** [comment]
Thank you for finding that @musicEnfanthen. They were disallowed within a `syllable` because we asked for that. We have found new cases where clef changes do occur within a syllable as shown above in the examples.

As for for the semantic difference, if `clef` (or `accid`) was only allowed outside, it would apply to the entire syllable and not be able to change the pitch of the neumes within a syllable.

**Comment by fujinaga** [comment]
After discussing this issue with some members of the @music-encoding/ig-neumes (if I post this on the mailing list, most people would have no idea what I am talking about) and in view of finding new notational practices (as shown above), I believe what we would like is the following elements to be allowed inside or outside of `syllable`:  `accid`, `divLine`, and `clef`.

As for `custos,` they should not be allowed inside a `syllable` and only outside.

**Comment by rettinghaus** [comment]
You want to treat an accidental like a free floating glyph, but it is always connected to a pitch event, so from my point of view it should be (always and only) a child of `nc`.

**Comment by rettinghaus** [comment]
A current problem I see is that [model.syllablePart](https://music-encoding.org/guidelines/dev/model-classes/model.syllablePart.html) is member of [model.eventLike](https://music-encoding.org/guidelines/dev/model-classes/model.eventLike.html), allowing as syllable children:
* [barLine](https://music-encoding.org/guidelines/dev/elements/barLine.html)
* [chord](https://music-encoding.org/guidelines/dev/elements/chord.html)
* [clef](https://music-encoding.org/guidelines/dev/elements/clef.html)
* [clefGrp](https://music-encoding.org/guidelines/dev/elements/clefGrp.html)
* [custos](https://music-encoding.org/guidelines/dev/elements/custos.html)
* [note](https://music-encoding.org/guidelines/dev/elements/note.html)
* [pad](https://music-encoding.org/guidelines/dev/elements/pad.html)
* [rest](https://music-encoding.org/guidelines/dev/elements/rest.html)
* [space](https://music-encoding.org/guidelines/dev/elements/space.html)

The only thing of this whole list that should go in there is `clef`, correct?

**Comment by fujinaga** [comment]
> A current problem I see is that [model.syllablePart](https://music-encoding.org/guidelines/dev/model-classes/model.syllablePart.html) is member of [model.eventLike](https://music-encoding.org/guidelines/dev/model-classes/model.eventLike.html), allowing as syllable children:
> 
> * [barLine](https://music-encoding.org/guidelines/dev/elements/barLine.html)
> * [chord](https://music-encoding.org/guidelines/dev/elements/chord.html)
> * [clef](https://music-encoding.org/guidelines/dev/elements/clef.html)
> * [clefGrp](https://music-encoding.org/guidelines/dev/elements/clefGrp.html)
> * [custos](https://music-encoding.org/guidelines/dev/elements/custos.html)
> * [note](https://music-encoding.org/guidelines/dev/elements/note.html)
> * [pad](https://music-encoding.org/guidelines/dev/elements/pad.html)
> * [rest](https://music-encoding.org/guidelines/dev/elements/rest.html)
> * [space](https://music-encoding.org/guidelines/dev/elements/space.html)
> 
> The only thing of this whole list that should go in there is `clef`, correct?

Yes, from the list the only thing we would like to be included within `syllable` is `clef`in the Neume module. I guess you can remove the others from [model.syllablePart](https://music-encoding.org/guidelines/dev/model-classes/model.syllablePart.html) because the model class is only used in the Neume module.
But we would also like to have `accid` and `divLine` to be allowed inside `syllable`.

**Comment by martha-thomae** [comment]
> You want to treat an accidental like a free floating glyph, but it is always connected to a pitch event, so from my point of view it should be (always and only) a child of `nc`.

There was a reason why `<accid>` is not allowed as a child of `<nc>`, see https://github.com/music-encoding/music-encoding/issues/816#issuecomment-861793579. But it seems to have been allowed within syllable at some point, as indicated in https://github.com/music-encoding/music-encoding/issues/665#issue-620153531.

**Comment by th-we** [comment]
> You want to treat an accidental like a free floating glyph, but it is always connected to a pitch event, so from my point of view it should be (always and only) a child of `nc`.

I see two reasons why the `<accid>` should be free-floating:

* Because it's spacial (horizontal) position must be encodable. It is in general not written immediately before the `<nc>` but before a `<ncGrp>` (or `<neume>`) which the `<nc>` is part of.  That's what @pe-ro's comment that @martha-thomae links to above says.
* I remember cases where the Corpus monodicum editors found an accidental that was written before a `<uneume>` (as it was encoded back then) that did not fit any of the `<note>`s (as they were encoded back then) from that `<uneume>`.  We didn't have a way of encoding it properly and used an annotation instead that my typesetting engine would recognize and output the proper accidental in the PDF. I would have to work through old backups to see if I find any traces of that, though.
* I also remember vaguely cases where editors were not sure if the accidentals they looked at were intended as something like a key signature or if they were just free-floating accidentals.  It would be nice if they could just encode what they find instead of enforcing an interpretation.

**Comment by fujinaga** [comment]
@yinanazhou can you create an MEI file that has an accid, a divLine, and a clef inside a syllable and also an accid, a divLine, and a clef outside a syllable and place that file in one of our DDMAL GitHub repo? This file will be used to test the changes we will be making here.

**Comment by yinanazhou** [comment]
> @yinanazhou can you create an MEI file that has an accid, a divLine, and a clef inside a syllable and also an accid, a divLine, and a clef outside a syllable and place that file in one of our DDMAL GitHub repo? This file will be used to test the changes we will be making here.

Hi @fujinaga, I've uploaded the test file [here](https://github.com/DDMAL/Neon/blob/develop/test/resources/schema_test.mei). The corresponding image file can be found in the same folder.

**Comment by ahankinson** [comment]
@yinanazhou @fujinaga could you please test your file against the latest changes in PR #983 and see if it solves your problem?

**Comment by yinanazhou** [comment]
Hi @ahankinson, I just tested an example file against this schema. I'm sorry that it doesn't solve all the problems as the elements below should be valid:
1. `accid` is not allowed either inside or outside of a syllable in this schema (`accid` should be allowed both inside and outside of a syllable)

- ![image](https://user-images.githubusercontent.com/74484576/185643968-11e21e0d-724f-45ec-aefb-cdec45fde808.png)
- ![image](https://user-images.githubusercontent.com/74484576/185644254-1369c4c9-54f7-497b-9bcf-6d380a47c614.png)

2. `divLine` is not allowed outside of a syllable in this schema (`divLine` should be allowed inside and outside of a syllable)

- ![image](https://user-images.githubusercontent.com/74484576/185644484-90b2aef1-47ab-4f97-8aa4-cf03fef638c1.png)

**Comment by ahankinson** [comment]
Just an observation, and not really related to the validity, but your `accid` should have a `ploc` attribute; otherwise you don't know which pitch on the staff your accidental affects.

**Comment by martha-thomae** [comment]
So, Klaus's PR #983 does solve the issues with `clef` and `custos`. Thank you, @rettinghaus!

I am providing here a very minimalistic MEI file to test any of the other desired changes mentioned in [this comment](https://github.com/music-encoding/music-encoding/issues/968#issuecomment-1146583421):

> ...I believe what we would like is the following elements to be allowed inside or outside of `syllable`: `accid`, `divLine`, and `clef`.
> 
> As for `custos,` they should not be allowed inside a `syllable` and only outside.


**File:** [testing_changes_to_neumes_module.xml.zip](https://github.com/music-encoding/music-encoding/files/9593018/testing_changes_to_neumes_module.xml.zip)
```xml
<?xml version="1.0" encoding="UTF-8"?>
<!-- <?xml-model href="https://music-encoding.org/schema/dev/mei-Neumes.rng" type="application/xml" schematypens="http://relaxng.org/ns/structure/1.0"?>
<?xml-model href="https://music-encoding.org/schema/dev/mei-Neumes.rng" type="application/xml" schematypens="http://purl.oclc.org/dsdl/schematron"?> -->
<?xml-model href="mei-Neumes.rng" type="application/xml" schematypens="http://relaxng.org/ns/structure/1.0"?>
<?xml-model href="mei-Neumes.rng" type="application/xml" schematypens="http://purl.oclc.org/dsdl/schematron"?>
<mei xmlns="http://www.music-encoding.org/ns/mei">
    <meiHead>
        <fileDesc>
            <titleStmt>
                <title>Testing Changes to Neumes Module</title>
            </titleStmt>
            <pubStmt/>
        </fileDesc>
    </meiHead>
    <music>
        <body>
            <mdiv>
                <score>
                    <scoreDef>
                        <staffGrp>
                            <staffDef n="1" lines="4" notationtype="neume"></staffDef>
                        </staffGrp>
                    </scoreDef>
                    <section>
                        <staff>
                            <layer>
                                <custos/>
                                <sb/>
                                <clef/>
                                <accid/>
                                <syllable>
                                    <custos/>
                                    <sb/>
                                    <clef/>
                                    <accid/>
                                    <neume><nc/></neume>
                                    <divLine/>
                                </syllable>
                                <divLine/>
                            </layer>
                        </staff>
                    </section>
                </score>
            </mdiv>
        </body>
    </music>
</mei>
```

**Comment by musicEnfanthen** [comment]
Reopened, since #983 only addressed parts of the issue.

**Comment by pe-ro** [comment]
I believe the fixes in PR #1019 resolve this issue.

However, it's SUPER important to make a distinction between what's allowed by MEI-all, which exists to accommodate documents that mix notational repertoires, and what's allowed by the neumes customization, which can extend or restrict MEI-all. For example, @fujinaga's  comments above -- "the only thing we would like to be included within syllable is clef in the Neume module." -- should be implemented in the neumes customization, not MEI-all; that is, not within the neumes module.
